lib/srv/termmanager.go

```diff
@@ -70,17 +71,9 @@ func NewTermManager() *TermManager {
 
 func (g *TermManager) writeToClients(p []byte) int {
 	g.lastWasBroadcast = false
-	truncateFront := func(slice []byte, max int) []byte {
-		if len(slice) > max {
-			return slice[len(slice)-max:]
-		}
-
-		return slice
-	}
-
-	g.history = append(g.history, truncateFront(p, maxHistory)...)
-	if len(g.history) > maxHistory {
-		g.history = g.history[:maxHistory]
+	g.history = append(g.history, truncateFront(p, maxHistoryBytes)...)
+	if len(g.history) > maxHistoryBytes {
+		g.history = g.history[:maxHistoryBytes]
```

Comment:
If I'm reading this right we append to the history, and then truncate it from the end? So once we reach `maxHistoryBytes` we would always append and then immediately truncate off what was just appended, always saving the oldest data and discarding new bytes, is that right?

Reply:
Yup, this keeps the lastest N bytes rolling with a maximum of 2N memory usage since it truncates the appendee from the front before appending it.

A test could be written to check that the length doesn't go past N after the fact. Sadly though that seems more like noise since length doesn't have to correspond to the size of the underlying array and thus the actual memory usage in Go. If you know of a better way to test that, please let me know.

Reply:
yeah looks right now, I'm not too worried about the memory contention just wanted to make sure the truncation was right